rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -178,6 +190,11 @@ TimeSource::~TimeSource()
   {
     this->detachNode();
   }
+
+  if (clock_executor_thread_.joinable()) {
```

Comment:
Sorry i just noticed that this is actually done in `TimeSource::detachNode`, so this statement should be deleted.

Reply:
You're right, I forgot to remove it from destructor. Actually, this statement is also in function `destroy_clock_sub()`:
```c
void TimeSource::destroy_clock_sub()
{
  std::lock_guard<std::mutex> guard(clock_sub_lock_);
  clock_subscription_.reset();
  if (clock_executor_thread_.joinable()) {
    clock_executor_.cancel();
    clock_executor_thread_.join();
    clock_executor_.remove_callback_group(clock_callback_group_);
  }
}
```
and function `detachNode()` actually just replicates what's in `destroy_clock_sub()`. 
So maybe the cleaner way is to:
- keep  `destroy_clock_sub()` as it currently is
- call  `destroy_clock_sub()` from `detachNode()` (and remove redundant statements)
- still call `detachNode()` from destructor  (and remove redundant statements)

Are you ok with this?

Reply:
i think that this is correct to do. i believe that `destroy_clock_sub` should be called when `detachNode` is called, since `Node` related objects will be reset. (this is a bug in current implementation.)